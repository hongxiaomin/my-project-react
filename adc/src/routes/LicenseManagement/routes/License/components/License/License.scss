.top-left{
  display: flex;
  justify-content: space-between;
  height: 50px;
  width: 100%;
  line-height: 50px;
  font-size: 16px;
  padding: 0px 30px;
  background-color: #f7f7f7;
  // border: 1px solid #d7d7d7;
  border-radius: 4px;
  margin-top: 20px;

}
.licenseCard {
  margin-top: 24px;
  padding: 30px;
  padding-top: 0px;
  border: 1px solid #ccc;
  overflow: auto;
  height: 490px;
  border-radius: 6px;
}
.button {
  height: 30px;
  margin-top: 10px;
}
.top {
  display: flex;
  justify-content: flex-start;
}
.model {
  margin-right: 10px;
  padding-top: 5px;
}
