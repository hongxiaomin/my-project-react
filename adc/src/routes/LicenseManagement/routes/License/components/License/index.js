import LicenseManagement from './License'

export default LicenseManagement
