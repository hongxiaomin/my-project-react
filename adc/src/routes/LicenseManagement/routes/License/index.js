import { injectReducer } from 'store/reducers'

export default store => ({
  path: 'licenseManagement',
  getComponent(nextState, cb) {
    /*  Webpack - use 'require.ensure' to create a split point
        and embed an async module loader (jsonp) when bundling   */
    require.ensure([], (require) => {
      /*  Webpack - use require callback to define
          dependencies for bundling   */
      const LicenseManagement = require('./components/License').default
      const re = require('./modules').default

      /*  Add the reducer to the store on key 'counter'  */
      injectReducer(store, { key: 'licenseManagement', reducer: re })

      /*  Return getComponent   */
      cb(null, LicenseManagement)

    /* Webpack named bundle   */
    }, 'licenseManagement')
  },
})
