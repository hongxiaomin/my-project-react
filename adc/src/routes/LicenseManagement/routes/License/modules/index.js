/* eslint no-unused-expressions: 0 */
import { assign } from 'lodash'
import { showMessage } from 'Utils'
import { simpleFetch } from 'helpers/fetchWrapper'

// ------------------------------------
// Constants
// ------------------------------------
export const GET_LICENSE_LIST = 'GET_LICENSE_LIST'
export const GET_USER_EMAIL = 'GET_USER_EMAIL'
export const GET_USER_LIST = 'GET_USER_LIST'
export const CHANGE_STATUS = 'CHANGE_STATUS'
export const UPDATE_LICENSE_DEVICEGPUS = 'UPDATE_LICENSE_DEVICEGPUS'
// ------------------------------------
// Async actions
// ------------------------------------


export const getLicenseList = () => dispatch =>
  // fetch('/api/ojs/licenseList', {
  simpleFetch('devices')
  .then((res) => {
    if (res.code === 200000) {
      dispatch({
        type: GET_LICENSE_LIST,
        payload: JSON.parse(res.result),
      })
    }
  })

/**
 * @param {id, trainingOrJudging} obj
 */
export const trainingOrJudging = obj => (dispatch, getState) => {
  return simpleFetch('device/op_mode', {
    method: 'PUT',
    body: obj,
  })
    .then((data) => {
      if (data.code !== 200000) {
        showMessage({ type: 'warning', text: data.message })
        const { licenseManagement: { licenseList } } = getState()
        const list = licenseList.map((item) => {
          if (item.id === obj.id * 1) {
            return Object.assign({}, item, { isTrainingDevice: !obj.isTrainingDevice })
          }
          return item
        })
        dispatch({
          type: GET_LICENSE_LIST,
          payload: list,
        })
      } else {
        showMessage({ type: 'success', text: data.message })

        // 根据返回值更新列表
        const { licenseManagement: { licenseList } } = getState()
        const list = licenseList.map((item) => {
          if (item.id === obj.id * 1) {
            return Object.assign({}, item, { isTrainingDevice: obj.isTrainingDevice })
          }
          return item
        })
        dispatch({
          type: GET_LICENSE_LIST,
          payload: list,
        })
      }
    })
}

/**
 * @param {*} obj id, deviceName
 * @param {*} cb Func
 */
export const save = (obj, cb) => (dispatch, getState) => {
  return simpleFetch('device/name', {
    method: 'PUT',
    body: obj,
  })
    .then((data) => {
      if (data.code !== 200000) {
        showMessage({ type: 'warning', text: data.message })
        return
      }
      if (data.code === 200000) {
        showMessage({ type: 'success', text: data.message })
      }
      if (cb) cb()
      // 根据返回值更新列表
      const { licenseManagement: { licenseList } } = getState()
      const list = licenseList.map((item) => {
        if (item.id === obj.id * 1) {
          return Object.assign({}, item, { name: obj.name })
        }
        return item
      })
      dispatch({
        type: GET_LICENSE_LIST,
        payload: list,
      })
    })
}

/**
 * 激活
 * @param {*} obj id, license
 */
export const activate = obj => (dispatch, getState) => {
  return simpleFetch('license/activate', {
    method: 'PUT',
    body: obj,
  })
    .then((data) => {
      if (data.code !== 200000) {
        showMessage({ type: 'warning', text: data.message })
        return
      }
      if (data.code === 200000) {
        showMessage({ type: 'success', text: data.message })
        const { licenseManagement: { licenseList } } = getState()
        const list = licenseList.map((item) => {
          if (item.id === obj.id * 1) {
            return Object.assign({}, item, { isActivate: true })
          }
          return item
        })
        dispatch({
          type: GET_LICENSE_LIST,
          payload: list,
        })
      }
    })
}

// get license
export const getLicense = obj => (dispatch, getState) => {
  return simpleFetch('license/serial_number', {
    method: 'POST',
    body: obj,
  })
    .then((data) => {
      if (data.code !== 200000) {
        showMessage({ type: 'warning', text: data.message })
        return
      }
      if (data.code === 200000) {
        showMessage({ type: 'success', text: data.message })

        const { licenseManagement: { licenseList } } = getState()
        const list = licenseList.map((item) => {
          if (item.id === obj.id * 1) {
            return Object.assign({}, item, { license: data.result })
          }
          return item
        })
        dispatch({
          type: GET_LICENSE_LIST,
          payload: list,
        })
      }
    })
}

export const getUserList = () => (dispatch) => {
  return simpleFetch('users')
    .then((data) => {
      if (data.code !== 200000) {
        showMessage({ type: 'warning', text: data.message })
        return
      }
      dispatch({
        type: GET_USER_LIST,
        data: data.result,
      })
    })
}

export const createUser = (postData, success, error) => (dispatch, getState) => {
  const { intl: { messages } } = getState()
  return simpleFetch('user', {
    method: 'POST',
    body: postData,
  })
    .then((data) => {
      if (data.code !== 200000) {
        showMessage({ type: 'error', text: data.message })
        error && error()
        return
      }
      showMessage({ text: messages.createSuccess })
      success && success()
      dispatch(getUserList())
    })
    .catch(() => {
      error && error()
    })
}

export const updateUser = (postData, success, error) => (dispatch, getState) => {
  const { intl: { messages } } = getState()
  return simpleFetch('user', {
    method: 'PUT',
    body: postData,
  })
    .then((data) => {
      if (data.code !== 200000) {
        showMessage({ type: 'error', text: data.message })
        error && error()
        return
      }
      success && success()
      showMessage({ text: messages.updateSuccess })
      dispatch(getUserList())
    })
}

export const deleteUser = userId => (dispatch, getState) => {
  const { intl: { messages } } = getState()
  return simpleFetch('user', {
    method: 'DELETE',
    body: { userId },
  })
    .then((data) => {
      if (data.code !== 200000) {
        showMessage({ type: 'error', text: data.message })
        return
      }

      const { licenseManagement: { userList } } = getState()
      const userListTmp = userList.filter(item => item.userId !== userId)
      showMessage({ text: messages.deleteSuccess })
      dispatch({
        type: GET_USER_LIST,
        data: userListTmp,
      })
    })
}

export const reload = () => dispatch =>
    // fetch('/api/ojs/licenseList', {
    simpleFetch('devices')
    .then((res) => {
      if (res.code === 200000) {
        dispatch({
          type: GET_LICENSE_LIST,
          payload: JSON.parse(res.result),
        })
        showMessage({ type: 'success', text: res.message })
      }
    })

export const getEmail = () => (dispatch) => {
  return simpleFetch('api/ojs/userEmail')
  .then((json) => {
    if (json.code === 200000) {
      dispatch({
        type: GET_USER_EMAIL,
        payload: json.result,
      })
      dispatch({
        type: CHANGE_STATUS,
        payload: true,
      })
    } else {
      dispatch({
        type: GET_USER_EMAIL,
        payload: json.result.data,
      })
    }
  })
}

const ACTION_HANDLERS = {
  [GET_LICENSE_LIST]: (state, action) => assign({}, state, {
    licenseList: action.payload,
  }),
  [GET_USER_EMAIL]: (state, action) => assign({}, state, {
    userEmail: action.payload,
  }),
  [CHANGE_STATUS]: (state, action) => assign({}, state, {
    status: action.payload,
  }),
  [GET_USER_LIST]: (state, action) => assign({}, state, {
    userList: action.data,
  }),
}


const initialState = {
  licenseList: [],
  userList: [],
  userEmail: '',
  status: false,
}

const licenseReducer = (state = initialState, action) => {
  const handler = ACTION_HANDLERS[action.type]

  return handler ? handler(state, action) : state
}

export default licenseReducer
