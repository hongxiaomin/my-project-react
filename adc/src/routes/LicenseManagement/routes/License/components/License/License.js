import React from 'react'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { injectIntl } from 'react-intl'
import { PropTypes } from 'prop-types'
import { connect } from 'react-redux'
import ReactTable from 'components/ADCTable'
import { getRole, getStorage } from 'Utils'
import Confirm from 'components/Confirm'
import LicenseCard from '../LicenseCard'
import Edit from '../Edit'
import classes from './License.scss'
import { getLicenseList, reload, getUserList, deleteUser } from '../../modules'

const { CreateUser, EditUser } = Edit
const Roles = getRole()

export class License extends React.Component {
  constructor() {
    super()
    this.state = {
      highlightedIndex: -1,
      totalPages: null,
      loading: true,
      isDialogOpen: false,

      editOpen: false,
      editDefautlValue: {},

      deleteOpen: false,
      deleteId: null,

      height: 'auto',
    }
  }

  componentDidMount() {
    this.props.dispatch(getLicenseList())
    this.props.dispatch(getUserList())
  }

  del = () => {
    this.setState({
      deleteOpen: false,
    })
    this.props.dispatch(deleteUser(this.state.deleteId))
  }

  edit = original => () => {
    this.setState({
      editOpen: true,
      editDefautlValue: original,
    })
  }

  toggleDialog = () => {
    this.setState({
      editOpen: !this.state.editOpen,
    })
  }
  reload = () => {
    this.props.dispatch(reload())
  }

  showDelConform = id => () => {
    this.setState({
      deleteOpen: !this.state.deleteOpen,
      deleteId: id,
    })
  }

  handleCancel = () => {
    this.setState({
      deleteOpen: false,
    })
  }

  handlePageSize = (pageSize) => {
    if (pageSize > 10) {
      this.setState({
        height: '500px',
      })
    }
  }

  render() {
    const { intl } = this.props

    return (
      <div style={{ margin: '30px 40px', display: 'flex', justifyContent: 'space-between' }}>
        <div style={{ width: '50%' }}>
          <div className={classes.top}>
            <h4 className={classes.model}><WrapperFormattedMessage id='userAndLicenseManagement.license.userManagement' /></h4>
            <CreateUser type="create" />
          </div>
          <div>
            <div className={classes['top-left']}>
              <span className={classes.productGlass}>
                <WrapperFormattedMessage id='userAndLicenseManagement.license.userList' />
              </span>
            </div>
            <ReactTable
              data={this.props.userList}
              defaultPageSize={10}
              onPageSizeChange={this.handlePageSize}
              className="-striped -highlight"
              style={{
                maxHeight: '500px',
                height: this.state.height,
              }}
              columns={[
                {
                  Header: intl.formatMessage({ id: 'userAndLicenseManagement.license.userID' }),
                  accessor: 'userId',
                  style: {
                    textAlign: 'center',
                  },
                },
                {
                  Header: intl.formatMessage({ id: 'userAndLicenseManagement.license.userName' }),
                  accessor: 'userName',
                  style: {
                    textAlign: 'center',
                  },
                },
                {
                  Header: intl.formatMessage({ id: 'role' }),
                  accessor: 'roleIds',
                  style: {
                    textAlign: 'center',
                  },
                  Cell: data => Roles[data.original.roleIds[0]],
                },
                {
                  Header: intl.formatMessage({ id: 'action' }),
                  width: 80,
                  accessor: '',
                  style: {
                    textAlign: 'center',
                    padding: '5px',
                  },
                  Cell: (data) => {
                    let delNode = (
                      <button
                        type='button'
                        style={{ cursor: 'pointer' }}
                        className={'pt-button pt-minimal pt-icon-delete'}
                        onClick={this.showDelConform(data.original.userId)}
                        title={intl.formatMessage({ id: 'delete' })}
                      />)
                    if (this.props.userInfo.userId === data.original.userId) {
                      delNode = (<span style={{
                        marginLeft: '10px',
                        display: 'inline-block',
                        width: 16,
                      }}
                      />)
                    }
                    return (<div>
                      <button
                        type='button'
                        style={{ cursor: 'pointer' }}
                        className={'pt-button pt-minimal pt-icon-edit'}
                        onClick={this.edit(data.original)}
                        title={intl.formatMessage({ id: 'edit' })}
                      />
                      {delNode}
                    </div>)
                  },
                },
              ]}
              getTrProps={() => ({
                style: {
                  height: '35px',
                  lineHeight: '20px',
                },
              })}
            />
          </div>
        </div>
        <div style={{ width: '45%' }}>
          <div className={classes.top}>
            <h4 className={classes.model}><WrapperFormattedMessage id='userAndLicenseManagement.license.licenseManagement' /></h4>
            <button
              style={{ width: '80px' }}
              type='button'
              onClick={this.reload}
              className={'pt-button pt-intent-primary'}
            >
              <WrapperFormattedMessage id='reload' />
            </button>
          </div>
          <div className={classes.licenseCard}>
            {
              this.props.licenseList.map(item => <LicenseCard item={item} key={item.id} />)
            }
          </div>
        </div>
        <EditUser
          isOpen={this.state.editOpen}
          toggleDialog={this.toggleDialog}
          defaultValue={this.state.editDefautlValue}
        />
        <Confirm
          isOpen={this.state.deleteOpen}
          title={intl.formatMessage({ id: 'userAndLicenseManagement.license.note' })}
          handleOk={this.del}
          handleCancel={this.handleCancel}
        />
      </div>
    )
  }
}

License.propTypes = {
  intl: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
  licenseList: PropTypes.array.isRequired,
  userList: PropTypes.array.isRequired,
  userInfo: PropTypes.object.isRequired,
}

export default connect((state) => {
  const str = getStorage('loginInfo')
  const userInfo = (str && JSON.parse(str)) || {}

  return {
    locale: state.intl.locale,
    token: state.auth.token,
    licenseList: state.licenseManagement.licenseList,
    userList: state.licenseManagement.userList,
    userInfo,
  }
})(injectIntl(License))
