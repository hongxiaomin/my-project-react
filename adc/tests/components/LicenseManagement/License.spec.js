import React from 'react'
import { shallow, mount } from 'enzyme'
import { License } from 'routes/LicenseManagement/routes/License/components/License/License'

describe('(Component) License', () => {
  const dispatch = jest.fn()
  const licenseList = []
  const userList = []
  const userInfo = {}
  it('render normally', () => {
    const wrapper = shallow(
      <License
        dispatch={dispatch}
        licenseList={licenseList}
        userList={userList}
        userInfo={userInfo}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})