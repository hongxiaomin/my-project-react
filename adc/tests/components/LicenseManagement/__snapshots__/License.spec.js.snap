// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) License render normally 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-between",
      "margin": "30px 40px",
    }
  }
>
  <div
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <div
      className="top"
    >
      <h4
        className="model"
      >
        User Management
      </h4>
      <Connect(CreateUser)
        type="create"
      />
    </div>
    <div>
      <div
        className="top-left"
      >
        <span
          className="productGlass"
        >
          User List
        </span>
      </div>
      <ReactTable
        AggregatedComponent={[Function]}
        ExpanderComponent={[Function]}
        FilterComponent={[Function]}
        LoadingComponent={[Function]}
        NoDataComponent={[Function]}
        PadRowComponent={[Function]}
        PaginationComponent={[Function]}
        PivotValueComponent={[Function]}
        ResizerComponent={[Function]}
        TableComponent={[Function]}
        TbodyComponent={[Function]}
        TdComponent={[Function]}
        TfootComponent={[Function]}
        ThComponent={[Function]}
        TheadComponent={[Function]}
        TrComponent={[Function]}
        TrGroupComponent={[Function]}
        aggregatedKey="_aggregated"
        className="-striped -highlight"
        collapseOnDataChange={true}
        collapseOnPageChange={true}
        collapseOnSortingChange={true}
        column={
          Object {
            "Aggregated": undefined,
            "Cell": undefined,
            "Expander": undefined,
            "Filter": undefined,
            "Footer": undefined,
            "Header": undefined,
            "Pivot": undefined,
            "PivotValue": undefined,
            "aggregate": undefined,
            "className": "",
            "filterAll": false,
            "filterMethod": undefined,
            "filterable": undefined,
            "footerClassName": "",
            "footerStyle": Object {},
            "getFooterProps": [Function],
            "getHeaderProps": [Function],
            "getProps": [Function],
            "headerClassName": "",
            "headerStyle": Object {},
            "minWidth": 100,
            "resizable": undefined,
            "show": true,
            "sortMethod": undefined,
            "sortable": undefined,
            "style": Object {},
          }
        }
        columns={
          Array [
            Object {
              "Header": "User ID",
              "accessor": "userId",
              "style": Object {
                "textAlign": "center",
              },
            },
            Object {
              "Header": "User Name",
              "accessor": "userName",
              "style": Object {
                "textAlign": "center",
              },
            },
            Object {
              "Cell": [Function],
              "Header": "Role",
              "accessor": "roleIds",
              "style": Object {
                "textAlign": "center",
              },
            },
            Object {
              "Cell": [Function],
              "Header": "Action",
              "accessor": "",
              "style": Object {
                "padding": "5px",
                "textAlign": "center",
              },
              "width": 80,
            },
          ]
        }
        data={Array []}
        defaultExpanded={Object {}}
        defaultFilterMethod={[Function]}
        defaultFiltered={Array []}
        defaultPageSize={10}
        defaultResized={Array []}
        defaultSortDesc={false}
        defaultSortMethod={[Function]}
        defaultSorted={Array []}
        expanderDefaults={
          Object {
            "filterable": false,
            "resizable": false,
            "sortable": false,
            "width": 35,
          }
        }
        filterable={false}
        freezeWhenExpanded={false}
        getLoadingProps={[Function]}
        getNoDataProps={[Function]}
        getPaginationProps={[Function]}
        getProps={[Function]}
        getResizerProps={[Function]}
        getTableProps={[Function]}
        getTbodyProps={[Function]}
        getTdProps={[Function]}
        getTfootProps={[Function]}
        getTfootTdProps={[Function]}
        getTfootTrProps={[Function]}
        getTheadFilterProps={[Function]}
        getTheadFilterThProps={[Function]}
        getTheadFilterTrProps={[Function]}
        getTheadGroupProps={[Function]}
        getTheadGroupThProps={[Function]}
        getTheadGroupTrProps={[Function]}
        getTheadProps={[Function]}
        getTheadThProps={[Function]}
        getTheadTrProps={[Function]}
        getTrGroupProps={[Function]}
        getTrProps={[Function]}
        groupedByPivotKey="_groupedByPivot"
        indexKey="_index"
        loading={false}
        loadingText="Loading..."
        nestingLevelKey="_nestingLevel"
        nextText="Next"
        noDataText="No rows found"
        ofText="of"
        onFetchData={[Function]}
        onPageSizeChange={[Function]}
        originalKey="_original"
        pageSizeOptions={
          Array [
            5,
            10,
            20,
            25,
            50,
            100,
          ]
        }
        pageText="Page"
        pivotDefaults={Object {}}
        pivotIDKey="_pivotID"
        pivotValKey="_pivotVal"
        previousText="Previous"
        resizable={true}
        rowsText="rows"
        showPageJump={true}
        showPageSizeOptions={true}
        showPagination={true}
        showPaginationBottom={true}
        showPaginationTop={false}
        sortable={true}
        style={
          Object {
            "height": "auto",
            "maxHeight": "500px",
          }
        }
        subRowsKey="_subRows"
      />
    </div>
  </div>
  <div
    style={
      Object {
        "width": "45%",
      }
    }
  >
    <div
      className="top"
    >
      <h4
        className="model"
      >
        License Management
      </h4>
      <button
        className="pt-button pt-intent-primary"
        onClick={[Function]}
        style={
          Object {
            "width": "80px",
          }
        }
        type="button"
      >
        Reload
      </button>
    </div>
    <div
      className="licenseCard"
    />
  </div>
  <Connect(EditUser)
    defaultValue={Object {}}
    isOpen={false}
    toggleDialog={[Function]}
  />
  <Confirm
    handleCancel={[Function]}
    handleOk={[Function]}
    isOpen={false}
    title="Are you sure to delete this user?"
  />
</div>
`;
